File: src/cviz/LayerState.java
package cviz;

/**
 * Track the layer state to deal with stopping on not the last frame
 */
public class LayerState {
	private final String videoName;
	private long lastFrame;
	
	public LayerState(String videoName) {
		this.videoName = videoName;
		lastFrame = 0;
	}

	public String getVideoName() {
		return videoName;
	}
	
	public long getLastFrame() {
		return lastFrame;
	}
	
	void setLastFrame(long lastFrame) {
		this.lastFrame = lastFrame;
	}

	public String toString() {
		return "LayerState: " + videoName + " " + lastFrame;
	}
}

File: CViz/LayerState.cs
namespace CViz
{
    enum LayerType
    {
        Video,
        Scene,
    }

    class LayerState
    {
        public string Name { get; }
        public long LastFrame { get; set; }
        public int LastFrameCount { get; set; }

        public LayerType Type { get; }

        public LayerState(LayerType type, string name)
        {
            Type = type;
            Name = name;
            LastFrame = 0;
            LastFrameCount = 0;
        }

        public override string ToString()
        {
            return $"LayerState: {Type} {Name} {LastFrame}({LastFrameCount})";
        }
    }
}
